### ISSUE TITLE
Production Logging

### ISSUE THREAD

**Issue by dciarniello** [spring-projects/spring-authorization-server#1111]
Production level logging is required to be able to easily troubleshoot production issues.

Logging was added in gh-159 but at the TRACE level.  TRACE is fine for development but not for production.

Please add logging more suitable for troubleshooting production issue.

**Comment by sjohnr** [comment]
@dciarniello, thanks for reaching out!

> Logging was added in https://github.com/spring-projects/spring-authorization-server/issues/159 but at the TRACE level. TRACE is fine for development but not for production.

The goal of gh-159 was to add enough logging to allow efficient troubleshooting during the development cycle without adding too much detail. Adding more details and more logging requires specific use cases and will not likely be addressed as a general request.

> Please add logging more suitable for troubleshooting production issue.

Please provide some additional information about your request, and be specific if possible. Depending on the number of use cases you can provide, we could potentially open an issue for each use case or an issue for one or more themes (groups of related use cases) that emerge from the request.

**Comment by sjohnr** [comment]
Thanks @dciarniello, that's a pretty good list to start with.

One thing we're thinking about right now (that is a bit of a detour from logging) is customization around the processing flow itself. I feel like if we add the right types of hooks into the processing flow of e.g. the `AuthenticationProvider`s or `Filter`s (such as in gh-1075 and a few other suggestions), there's potential to solve two problems at once. Meaning, even if the core framework didn't have direct support for every single logging case, it would ideally be trivial to add your own logging in a specific place without much work. This would help unblock some of those needs from being bound to a specific future release, if you need it sooner. So I see two things here:

1. Add hooks into the processing flow, specifically for failure cases.
2. Add additional logging in the framework itself.

Depending on where either of these fits in the priority list, one could be addressed earlier than the other. I also think that the second request (this issue) could be addressed as contributions from the community since we have some logging in place already to show the general (simple) pattern. So if you're interested in sending any PRs, let me know!

For now, I'll do some checking, and see about adding an issue for each of the logging use cases you point out above so they are small bite-sized issues, then I'll close this issue as a duplicate. Sound good? Let me know if you have any other thoughts.

**Comment by sjohnr** [comment]
Closing as a duplicate of gh-1245, gh-1246, gh-1247, and gh-1248.

**Comment by dciarniello** [comment]
Hi, @sjohnr.

Thanks for looking into this and creating the individual tickets.  Much appreciated.
